geojson_pydantic/features.py

```diff
@@ -29,9 +29,11 @@ class Config:
     @validator("geometry", pre=True, always=True)
     def set_geometry(cls, geometry: Any) -> Any:
         """set geometry from geo interface or input"""
-        if hasattr(geometry, "__geo_interface__"):
-            return geometry.__geo_interface__
-        return geometry
+        if geometry is not None:
+            if hasattr(geometry, "__geo_interface__"):
```

Comment:
This shouldn't be necessary `hasattr(None, "any_sting") == False`.

Reply:
yeah I'll revert this change. TBH I did add this test because I was also planning to fix #93 but that will wait